nntrainer/models/neuralnet.cpp

```diff
@@ -396,14 +403,16 @@ void NeuralNetwork::load(const std::string &file_path,
       /// read. so, after this line, additional read shouldn't be called
       model_file.seekg(bin_file_pos);
 
-      if (istrequal(opt->getType(), "adam")) {
-        char opt_type[4];
-        model_file.read(opt_type, 4);
-        if (istrequal(opt_type, "adam")) {
-          for (auto iter = model_graph.cbegin(); iter != model_graph.cend();
-               iter++) {
-            (*iter)->read(model_file, true);
-          }
+      opt->read(model_file);
+      unsigned int exist_op_var = 0;
+
+      checkedRead(model_file, (char *)&exist_op_var, sizeof(exist_op_var));
+
+      if (istrequal(opt->getType(), "adam") && exist_op_var) {
```

Comment:
I think 
`istrequal(opt->getType(),` "adam")`
part have to move line 415 like
(*iter)->read(model_file, true, istrequal(opt->getType(), "adam") && load_opt_var);

Think of the case which trained with adam and save optimizer variables and read with sgd optimizer.

Reply:
in the case of sgd and other optimizer, we could not load the variables because it does not have variables. so.. I guess we need save it always at the end of binary file. so that we could just finish read if the variables does not need.